client/src/featureform/providers/filestore.py

```diff
@@ -0,0 +1,139 @@
+
+    def __init_table(self):
+        self.df = pd.DataFrame(data={"entity": [], "value": []}).astype(
+            {"entity": "string", "value": self.stype}
+        )
+        self.df.to_csv(self.filepath, index=False)
+
+    def __init_update_df(self):
+        df = pd.DataFrame(columns=["entity", "value"]).astype(
+            {"entity": "str", "value": self.stype}
+        )
+        return df
+
+    def get_path(self):
+        return self.filepath
+
+    def set(self, key, value):
+        df = pd.DataFrame(data={"entity": [key], "value": [value]}).astype(
+            {"entity": "str", "value": self.stype}
+        )
+        self.df = self.df.append(df)
+
+    def get(self, key):
+        # The most efficient way to perform the upsert at the moment is to
+        # create an array with new values to be inserted then process them
+        # all at the same time in memory.
+        self.flush()
+        df = self.df
+        try:
+            value = df[df["entity"] == key]["value"].astype(self.stype).values[0]
+        except IndexError:
+            raise ValueError(
+                f"Could not find value for feature: {self.name} ({self.variant}) for: {key}"
+            )
+        if self.stype == "datetime64[ns]":
+            return pd.to_datetime(value)
+        return value
+
+    def flush(self):
+        self.df = (
+            pd.concat([self.df, self.update_df])
+            .drop_duplicates("entity", keep="last")
+            .sort_values("entity", ascending=False)
+            .reset_index(drop=True)
+        )
+        self.df.to_csv(self.filepath, mode="w", index=False, header=True)
+
+
+class LocalFileStore(OnlineStore):
+    def __init__(self):
+        self.path = ".featureform/features"
+        self.type_table = f"{self.path}/types.csv"
+        if not os.path.exists(self.path):
+            os.makedirs(self.path, exist_ok=True)
+
+    def create_table(self, name, variant, entity_type: ScalarType):
+        if os.path.exists(f"{self.path}/{name}/{variant}.csv"):
+            raise ValueError(f"Table {name} with variant {variant} already exists")
+
+        self.__set_type(name, variant, entity_type)
+
+        table = LocalFileTable(
+            name, variant, self.path, self.__convert_type(entity_type)
+        )
+        return table
+
+    def __set_type(self, name, variant, entity_type: ScalarType):
+        header = True
+        if os.path.exists(self.type_table):
+            header = False
+
+        types_df = pd.DataFrame(
+            data={"name": [name], "variant": [variant], "type": [entity_type.value]}
+        ).astype({"name": "string", "variant": "string", "type": "string"})
+
+        types_df.to_csv(self.type_table, mode="a", header=header, index=False)
+
+    def get_table(self, name, variant):
+        if not os.path.exists(f"{self.path}/{name}/{variant}.csv"):
+            raise ValueError(f"Table {name} with variant {variant} does not exist")
```

Comment:
I don't think we need this since self.__get_type will do this for you

Reply:
Would like to leave it. It differentiates whether there was a failure with getting the table itself, or if it was a failure with getting metadata around the table